cpp/log/cert_checker.cc

```diff
@@ -100,20 +102,17 @@ bool CertChecker::LoadTrustedCertificatesFromBIO(BIO* bio_in) {
     if (x509 != NULL) {
       // TODO(ekasper): check that the issuing CA cert is temporally valid
       // and at least warn if it isn't.
-      Cert* cert = new Cert(x509);
+      unique_ptr<Cert> cert(new Cert(x509));
       string subject_name;
       CertVerifyResult is_trusted = IsTrusted(*cert, &subject_name);
       if (is_trusted != OK && is_trusted != ROOT_NOT_IN_LOCAL_STORE) {
-        delete cert;
         error = true;
         break;
       }
 
       ++cert_count;
-      if (is_trusted == OK) {
-        delete cert;
-      } else {
-        certs_to_add.push_back(make_pair(subject_name, cert));
+      if (is_trusted != OK) {
```

Comment:
Yeah, the whole thing is a bit wacky. I didn't even change the `x509 != NULL` to a `!x509`, nor add any `const`, or all the rest of the things I usually do, but I was most frightened by the local heap allocation and all the branches...